### PR TITLE
Testing concepts

### PR DESCRIPTION
**Pull request by josuepuig-wz**
## Description
This is a PR for the testing concepts

## Goals
- [x]  Meaningful test cases were implemented for components and sub-routines logic.
- [x] All the test cases were successful.

### CODE REVIEW

File: src/components/Header/Header.test.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import App from '../App/App.component';
+
+describe('App Component Tests', () => {
+  const setState = jest.fn();
+  const useStateSpy = jest.spyOn(React, 'useState');
+  useStateSpy.mockImplementation((init) => [init, setState]);
+
+  beforeEach(() => {
+    render(<App />);
```
Comment (andresSaldanaAguilar) [inline src/components/Header/Header.test.jsx]: This should only render the `<Header/>` component to do unit testing on it, if you render the App component, you are rendering the whole app to this test suit and the main purpose here is to only test the header and avoid extra dependencies

File: src/pages/Secret/Secret.test.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
```
Comment (andresSaldanaAguilar) [inline src/pages/Secret/Secret.test.jsx:1]: 😄 nice testing haha

File: src/components/App/App.test.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  it('Should render the Home component', async () => {
+    expect(screen.getByTestId('home-component')).toBeInTheDocument();
+  });
+
+  it('Should change background color with a click', async () => {
+    const { body } = document;
+    const spy = jest.spyOn(body.style, 'setProperty');
+
+    body.click();
+    expect(spy).toHaveBeenCalled();
```
Comment (andresSaldanaAguilar) [inline src/components/App/App.test.jsx:23]: I couldn't figure out what is being tested here, can you explain to me?

**Comment by andresSaldanaAguilar** [comment]
## Review (Mini challenge 2)
Good Job Josue, I like how you used the Jest library, as a general comment, I saw test cases for pages/views but I didn't see tests for components, so let's have them too.
Also, I couldn't run the test because netlify is just for hosting, so If you can take a screenshot now or for further deliverables of tests and coverage would be very helpful to me to see them running.
And don't forget to use snapshots, they help to state how does the components should look like.

## Acceptance Criteria (Mini challenge 2)
- [x] Meaningful test cases were implemented for components and sub-routines logic
- [x] All the test cases were successful: couldn't run tests, but they seem right

## Bonus Points ( Mini challenge 2)
- [ ] Test coverage is above 60% : coverage report not found

File: src/components/App/App.test.jsx (hunk 3 above)
Comment (josuepuig-wz) [reply src/components/App/App.test.jsx:23]: Yes, this was to test that the useEffect in App component had been called. The one that changes the colour background

File: src/components/Header/Header.test.jsx (hunk 1 above)
Comment (josuepuig-wz) [reply src/components/Header/Header.test.jsx]: Changed

File: src/pages/Secret/Secret.test.jsx (hunk 2 above)
Comment (josuepuig-wz) [reply src/pages/Secret/Secret.test.jsx:1]: Thanks haha

**Pull request merged**